vignettes/irecist.Rmd

```diff
@@ -0,0 +1,877 @@
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(ADT, RSSEQ),
+    new_var = ANL02FL,
+    condition = AVALC == "iCPD",
+    mode = "first",
+    selection = "before",
+    inclusive = TRUE
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, AVALC, ADT, ANL01FL, ANL02FL)
+)
+```
+
+### Select Source Assessments for Parameter derivations
+
+For most parameter derivations the post-baseline overall response assessments up
+to and including first iCPD are considered.
+```{r}
+ovr <- filter(adrs, PARAMCD == "OVR" & ANL01FL == "Y" & ANL02FL == "Y")
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  ovr,
+  display_vars = exprs(USUBJID, AVISIT, AVALC, ADT, RANDDT)
+)
+```
+
+## Define Events
+
+The building blocks for the events that contribute to deriving common endpoints
+like what constitutes a responder, or a Best Overall Response of complete
+response (CR), ... are predefined in admiralonco for Recist 1.1 (see [Pre-Defined
+Response Event Objects](../reference/event_objects.html)). New Response Event Objects 
+are needed for iRECIST and any study-specific needs. 
+
+```{r}
+icpd_y <- event_joined(
+  description = paste(
+    "Define confirmed progressive disease (iCPD) as",
+    "iUPD followed by iCPD with only other iUPD and NE responses in between"
+  ),
+  dataset_name = "ovr",
+  join_vars = exprs(AVALC, ADT),
+  join_type = "after",
+  first_cond_upper = AVALC.join == "iCPD",
+  condition = AVALC == "iUPD" &
+    all(AVALC.join %in% c("iCPD", "iUPD", "NE" )),
+  set_values_to = exprs(AVALC = "Y")
+)
+
+iupd_y <- event_joined(
+  description = paste(
+    "Define unconfirmed progressive disease (iUPD) as",
+    "iUPD followed only by other iUPD or NE responses"
+  ),
+  dataset_name = "ovr",
+  join_vars = exprs(AVALC, ADT),
+  join_type = "all",
+  condition = ADT <= ADT.join & AVALC == "iUPD" & all(AVALC.join %in% c("iUPD", "NE")),
+  set_values_to = exprs(AVALC = "Y")
+)
+
+no_data_n <- event(
+  description = "Define no response for all patients in adsl (should be used as last event)",
+  dataset_name = "adsl",
+  condition = TRUE,
+  set_values_to = exprs(AVALC = "N"),
+  keep_source_vars = adsl_vars
+)
+
+no_data_missing <- event(
+  description = paste(
+    "Define missing response (MISSING) for all patients in adsl (should be used",
+    "as last event)"
+  ),
+  dataset_name = "adsl",
+  condition = TRUE,
+  set_values_to = exprs(AVALC = "MISSING"),
+  keep_source_vars = adsl_vars
+)
+
+```
+
+## Derive Confirmed and Unconfirmed Progressive Disease Parameter {#pd}
+
+Now that we have the input records prepared above with any
+company-specific requirements, we can start to derive new parameter
+records. For the parameter derivations, all values except those
+overwritten by `set_values_to` argument are kept from the earliest
+occurring input record fulfilling the required criteria.
+
+When an `iCPD` occurs,the date would be that of the first `iUPD`. 
```

Comment:
add a space in "occurs,the"
I would modify it a bit:
"When an iCPD occurs, the date of progression would be the same as RECIST 1.1 date (i.e. first iUPD date in that block/bar)."

Reply:
updated